src/coreclr/gc/gcinterface.h

```diff
@@ -453,20 +453,6 @@ typedef enum
      *
      */
     HNDTYPE_SIZEDREF     = 8,
-
-    /*
-     * NATIVE WEAK HANDLES
-     *
-     * Native weak reference handles hold two different types of weak handles to any
-     * RCW with an underlying COM object that implements IWeakReferenceSource.  The
-     * object reference itself is a short weak handle to the RCW.  In addition an
-     * IWeakReference* to the underlying COM object is stored, allowing the handle
-     * to create a new RCW if the existing RCW is collected.  This ensures that any
-     * code holding onto a native weak reference can always access an RCW to the
-     * underlying COM object as long as it has not been released by all of its strong
-     * references.
-     */
-    HNDTYPE_WEAK_NATIVE_COM   = 9
```

Comment:
GC/VM interface change. cc @dotnet/gc

Reply:
this changes looks ok, assuming this enum is not shared between clrgc and coreclr? We need to ensure that latest clrgc continues to work with .net 7 runtime.

Reply:
Everything in gcinterface.h is shared between clrgc and coreclr.

> We need to ensure that latest clrgc continues to work with .net 7 runtime.

The changes under src/coreclr/gc need to be reverted if you would like to maintain this invariant.

Reply:
> continues to work with .net 7 runtime

Is this a temporary requirement?
Have we already revved the API version for the frozen string literals work?

Reply:
We have added a new method to IGCHeap as part of the frozen literals work, that does not prevent new clrgc + old coreclr from working.

Reply:
ok. Considering how much GC and VM are coupled, it feels like compat requirement like this is not sustainable for long. Hopefully this is just for the short term.

Reply:
I don't see why that's not sustainable. we just don't delete things from the interface. yes, it'd be a bit nice to be able to delete things but considering the benefit it seems like the right tradeoff.